homeassistant/components/squeezebox_player/media_player.py

```diff
@@ -0,0 +1,218 @@
+            name=self.player.name,
+            hw_version=self.player.firmware,
+        )
+        # PiCore player has web interface
+        if "-pCP" in self.player.firmware:
+            self._attr_device_info[
+                "configuration_url"
+            ] = f"http://{self.player.device_address}"
+
+    async def async_media_play(self) -> None:
+        """Send play command to device."""
+        await self.player.play()
+
+    async def async_media_pause(self) -> None:
+        """Send pause command to device."""
+        await self.player.pause()
+
+    async def async_media_stop(self) -> None:
+        """Send stop command to device."""
+        await self.player.stop()
+
+    async def async_set_volume_level(self, volume: float) -> None:
+        """Send new volume_level to device."""
+        volume = int(volume * 100)
+        await self.player.volume_set(volume)
+
+    async def async_mute_volume(self, mute: bool) -> None:
+        """Mute the volume."""
+        await self.player.mute(mute)
+
+    async def async_turn_on(self) -> None:
+        """Turn on device."""
+        await self.player.power(True)
+
+    async def async_turn_off(self) -> None:
+        """Turn off device."""
+        await self.player.power(False)
+
+    async def async_play_media(self, media_type: str, media_id: str, **kwargs) -> None:
+        """Send the play_media command to the media player."""
+        # Handle media_source
+        if media_source.is_media_source_id(media_id):
+            sourced_media = await media_source.async_resolve_media(self.hass, media_id)
+            media_id = sourced_media.url
+            media_type = sourced_media.mime_type
+
+        if not media_type.startswith("audio/"):
+            media_type = None  # type: ignore[assignment]
+        await self.player.play_url(media_id, mime_type=media_type)
+
+    async def async_browse_media(
+        self, media_content_type=None, media_content_id=None
+    ) -> BrowseMedia:
+        """Implement the websocket media browsing helper."""
+        return await media_source.async_browse_media(
+            self.hass,
+            media_content_id,
+            content_filter=lambda item: item.media_content_type.startswith("audio/"),
+        )
+
+    async def _on_slim_event(self, event: SlimEvent):
+        """Call when we receive an event from SlimProto."""
+        if event.type == EventType.PLAYER_CONNECTED:
+            # player reconnected, update our player object
+            self.player = self.slimserver.get_player(event.player_id)
+        if event.type == EventType.PLAYER_RPC_EVENT:
+            # rpc event from player such as a button press,
+            # forward on the eventbus for others to handle
+            evt_data = {
+                **event.data,
+                "entity_id": self.entity_id,
+                "player_id": event.player_id,
+            }
+            self.hass.bus.async_fire(PLAYER_EVENT, evt_data, EventOrigin.remote)
```

Comment:
`remote` origin is only if people fire events via the API. This is fired from inside HA because of an external non-HA event, so that's local 🙃 

```suggestion
            self.hass.bus.async_fire(PLAYER_EVENT, evt_data)
```

Reply:
ah, i wasn't completely sure about the meaning of eventorigin. I guessed wrong :-)